src/Apps/Order/Components/OfferHistoryItem.tsx

```diff
@@ -1,93 +1,61 @@
-      ({ node: { id } }) => id !== lastOffer.id
-    )
-    return (
-      <StepSummaryItem {...others}>
-        <Row>
-          <Serif size={["2", "3"]} color="black60">
-            {lastOffer.fromParticipant === "SELLER"
-              ? "Seller's offer"
-              : "Your offer"}
-          </Serif>
-          <Serif size={["2", "3"]} color="black100">
-            {lastOffer.amount}
-          </Serif>
-        </Row>
-        <Row mb={1}>
-          <div />
-          <Sans size="2" color="black60">
-            List price: {totalListPrice}
-          </Sans>
-        </Row>
-        {previousOffers.length > 0 && (
-          <Flex
-            flexDirection="column"
-            position="relative"
-            style={{ minHeight: "26px" }}
-          >
-            <Row
-              position="absolute"
-              style={{
-                right: "0",
-                transition: "opacity 0.24s ease",
-                opacity: showingFullHistory ? 0 : 1,
-                pointerEvents: showingFullHistory ? "none" : "all",
-              }}
-            >
-              <div />
-              <Button
-                variant="secondaryGray"
-                size="small"
-                onClick={() => {
-                  this.setState({ showingFullHistory: true })
-                }}
-              >
-                Show offer history
-              </Button>
-            </Row>
-            <StaticCollapse open={showingFullHistory}>
-              <Flex m={0} flexDirection="column">
-                {previousOffers.map(({ node: offer }) => (
-                  <Row key={offer.id}>
-                    <Serif size={["2", "3"]} color="black60">
-                      {offer.fromParticipant === "BUYER" ? "You" : "Seller"}
-                      {` (${offer.createdAt})`}
-                    </Serif>
-                    <Serif size={["2", "3"]} color="black60">
-                      {offer.amount}
-                    </Serif>
-                  </Row>
-                ))}
-              </Flex>
-            </StaticCollapse>
-          </Flex>
-        )}
-      </StepSummaryItem>
-    )
-  }
+  } & StepSummaryItemProps
+> = ({ order: { totalListPrice, lastOffer, offers }, ...others }) => {
+  const previousOffers = offers.edges.filter(
+    ({ node: { id } }) => id !== lastOffer.id
+  )
+  return (
+    <StepSummaryItem {...others}>
+      <Row>
+        <Serif size={["2", "3"]} color="black60">
+          {lastOffer.fromParticipant === "SELLER"
+            ? "Seller's offer"
+            : "Your offer"}
+        </Serif>
+        <Serif size={["2", "3"]} color="black100">
+          {lastOffer.amount}
+        </Serif>
+      </Row>
+      <Row>
+        <div />
+        <Sans size="2" color="black60">
+          List price: {totalListPrice}
+        </Sans>
+      </Row>
+      {previousOffers.length > 0 && (
+        <>
+          <Spacer mb={1} />
+          <RevealButton buttonLabel="Show offer history">
```

Comment:
This file changed because I factored out the 'click button to reveal more stuff' pattern.